[WIP] user Signup -feature #101

Gerald1614 opened the pull request:
implementation of user signup.  had to deploy UUID to generate user id (neo4j ids are not well managed) , used middleware for password and slug, used a fake image for the avatar. It was tested in graphql playground, seems to work as planned. The function returns the same object as when a user loggedIn so we can log them directly. the UI will have to be built as well as the logic to prevent those unverified users to access features they are not supposed to access while their emails are still not validated. I also checked if user already existed but with the email as I think the email uniqueness is more accurate than names. please send your feedback on this iteration.
Also I may have generate errors in the login test as I decided to change the names of the mutations, using uppercase on the first letter of the Login and Signup resolvers as it is the case for all mutations. I am sorry for that.

appinteractive committed:
Improved locale handling and moved MAPBOX_TOKEN to env

appinteractive committed:
Throw an error if requried env vars are missing

appinteractive committed:
Fixed login and schema

appinteractive committed:
Fixed injection issues

appinteractive committed:
Fixed short_code

appinteractive committed:
Fixed eslint issue

appinteractive committed:
Internationalized Locations

appinteractive committed:
Fixed issues for non german / english languages

appinteractive committed:
Fixed user middleware

appinteractive committed:
Added locations to seeded users

Gerald1614 committed:
Signup user

Gerald1614 committed:
change login to Login in test (sync on nomencalture for resolvers names )

Gerald1614 committed:
change naomenclature for Login and Signup

Gerald1614 committed:
Merge branch 'master' into user_signup

Gerald1614 committed:
leveraging middleware for createdAt property

Gerald1614 committed:
Merge branch 'user_signup' of https://github.com/Gerald1614/Nitro-Backend into user_signup

appinteractive committed:
Refactoring

appinteractive commented on src/graphql-schema.js:
```diff
@@ -81,24 +82,47 @@ export const resolvers = {
       return session.run(
         'MATCH (user:User {email: $userEmail}) ' +
-        'RETURN user {.id, .slug, .name, .avatar, .email, .password, .role} as user LIMIT 1', { userEmail: email })
+        'RETURN user {.id, .slug, .name, .avatar, .email, .verified, .password, .role} as user LIMIT 1', { userEmail: email })
```
please remove password here

appinteractive commented on src/middleware/dateTimeMiddleware.js:
```diff
@@ -2,7 +2,7 @@ import format from 'date-fns/format'
 export default {
   Mutation: {
-    CreateUser: async (resolve, root, args, context, info) => {
+    Signup: async (resolve, root, args, context, info) => {
```
please create new entry as the CreateUser is needed for seeding and admin

Gerald1614 replied:
in fact if we create a CreateUSer mutation in the graphpql schema, graphql will use our resolver. so maybe it a better path to change the resolve to CreateUser instead of creating a Signup.
let me know what you think.

appinteractive replied:
Yeah I think maybe it’s better to use the create user for both. We just need to keep in mind that it might be used by the admin later.

appinteractive commented on src/middleware/passwordMiddleware.js:
```diff
@@ -3,7 +3,7 @@ import walkRecursive from './helpers/walkRecursive'
 export default {
   Mutation: {
-    CreateUser: async (resolve, root, args, context, info) => {
+    Signup: async (resolve, root, args, context, info) => {
```
please create new entry as the CreateUser is needed for seeding and admin

appinteractive commented on src/middleware/sluggifyMiddleware.js:
```diff
@@ -9,7 +9,7 @@ export default {
       return result
     },
-    CreateUser: async (resolve, root, args, context, info) => {
+    Signup: async (resolve, root, args, context, info) => {
```
please create new entry as the CreateUser is needed for seeding and admin

Gerald1614 committed:
removed psssword from return object and modified login test with new nomenclature (Login instead of login)

Gerald1614 committed:
removed psssword from return object and modified login test with new nomenclature (Login instead of login)

roschaefer committed:
Add missing env variable to `docker-compose.yml`

@appinteractive always run the docker-compose up workflow at least once.

roschaefer committed:
Quote JWT Token (consistency)

Gerald1614 committed:
changed Signup to CreateUser

Gerald1614 committed:
changed Signup to CreateUser

Gerald1614 commented:
@appinteractive I just updated the code so CreateUser is used instead of signup. I know you asked me to remove password from the object return on login too, bu tI put it back as we need it to verify the user password. but the UserLoggedIn schema is not exposing the password when we return the user to the app.  I know people were engaged in the tests, I do not know if you want to implement the tests for the CreateUser or if somebody more specialist could do that.

roschaefer commented:
@Gerald1614 I'm working to improve the testing pipeline, currently in #106 and yes, I would love to see a test for this :wink: we can work together on a test for this PR, if you like. Synchronously in a pair-programming or asynchronously.

appinteractive committed:
Try to fix build issues

appinteractive committed:
Merge branch 'master' into user_signup

appinteractive committed:
Merge branch 'master' into user_signup

appinteractive committed:
Fixed isOwner permission

roschaefer committed:
Fix seed data in docker environment

@appinteractive

Running
```
docker-compose -f ../Nitro-Backend/docker-compose.yml -f ../Nitro-Backend/docker-compose.travis.yml up
```
from the Nitro-Web folder gives me an error when running seed data.

The configuration in this commit fixes it on my machine. I guess that
will fix the frontend build, too.

roschaefer committed:
Fix all cypress tests but one

@appinteractive, it's super strange but apparently the JWT has to be
enclosed in quotes whereas the MAPBOX_TOKEN must not be enclosed. This
commit fixes almost all tests running in the docker containers on my
machine. The remaining test case is:
```
About me and and location
I set my location to "Mecklenburg-Vorpommern" (example #2)
```

@appinteractive when I try to repeat the cypress test on my machine I
get no results for "Mecklenburg-Vorpommern" only cities and states. can
you check if regions really work?

dependabot-support committed:
Bump supertest from 3.3.0 to 3.4.1

Bumps [supertest](https://github.com/visionmedia/supertest) from 3.3.0 to 3.4.1.
- [Release notes](https://github.com/visionmedia/supertest/releases)
- [Changelog](https://github.com/visionmedia/supertest/blob/master/History.md)
- [Commits](https://github.com/visionmedia/supertest/compare/v3.3.0...v3.4.1)

Signed-off-by: dependabot[bot] <support@dependabot.com>

dependabot-support committed:
Bump eslint from 5.11.1 to 5.12.0

Bumps [eslint](https://github.com/eslint/eslint) from 5.11.1 to 5.12.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/master/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v5.11.1...v5.12.0)

Signed-off-by: dependabot[bot] <support@dependabot.com>

dependabot-support committed:
Bump eslint-plugin-node from 8.0.0 to 8.0.1

Bumps [eslint-plugin-node](https://github.com/mysticatea/eslint-plugin-node) from 8.0.0 to 8.0.1.
- [Release notes](https://github.com/mysticatea/eslint-plugin-node/releases)
- [Commits](https://github.com/mysticatea/eslint-plugin-node/compare/v8.0.0...v8.0.1)

Signed-off-by: dependabot[bot] <support@dependabot.com>

Gerald1614 committed:
Merge branch 'user_signup' of https://github.com/Gerald1614/Nitro-Backend into user_signup

Gerald1614 commented:
@roschaefer ,i will be happy to work with you but I am entering now a crazy period as I need to close a few strategic deals before end of fiscal year (which ends on January 31th) i doubt I will be able to contribute a lot in between.

Lulalaby commented:
Please fix 
```
Conflicting files
package.json
src/graphql-schema.js
src/graphql-schema.test.js
src/middleware/permissionsMiddleware.js
src/schema.graphql
src/seed/data/index.js
src/seed/data/users.js
src/server.js
yarn.lock
```
via commandline

roschaefer commented:
@Gerald1614 we're migrating to a [monorepo](https://github.com/Human-Connection/Human-Connection). So many things have changed including how to write tests for resolvers that I hope you don't mind if we close this PR unmerged?

@Gerald1614 I would suggest that we meet on [Discord](https://discordapp.com/invite/6ub73U3) and we can start over on the signup feature. **It is** a very basic feature.
